File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    result = ''
    mode = 0
    
    for i in range(len(code)):
        if mode == 0:          # 모드가 0일 때
            if code[i] == '1':   # 모드바꾸기
                mode = 1
            elif i % 2 == 0 :     # idx가 짝수일 때
                result += code[i]
                
        else:                  # 모드가 1일 때
            if code[i] == '1':
                mode = 0
            elif i % 2 == 1:    # idx가 홀수일 때
                result += code[i]
                
    if len(result) == 0:
        return "EMPTY"
    else:
        return result
                

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        int mode = 0;
        String ret = "";
        for(int i=0; i<code.length(); i++) {
            char temp = code.charAt(i);
            
            if(temp == '1'){    //모드 바꾸기 
                if(mode == 0){
                    mode = 1;
                    continue;
                }else {
                    mode = 0;
                    continue;
                }
            }
            // 모드가 0일 때    
            if(mode == 0 && i%2 == 0){
                ret += temp;
                
            }else if(mode == 1 && i%2 == 1) {
                ret += temp;
            }
            
        }    
    return ret.isEmpty() ? "EMPTY": ret;
    }
}